uefi-test-runner/src/boot/mod.rs

```diff
@@ -36,3 +39,72 @@ fn test_locate_handle_buffer(bt: &BootServices) {
         );
     }
 }
+
+/// This test loads the "self image" again into memory using the `load_image`
+/// boot service function. The image is not started but just loaded into memory.
+///
+/// It transitively tests the protocol [`LoadedImageDevicePath`] which is
+/// required as helper.
+fn test_load_image(bt: &BootServices) {
+    info!("Testing the `load_image` function");
+
+    let image_device_path_protocol = bt
+        .open_protocol_exclusive::<LoadedImageDevicePath>(bt.image_handle())
+        .expect("should open LoadedImage protocol");
+
+    // Note: This is the full device path. The LoadedImage protocol would only
+    // provide us with the file-path portion of the device path.
+    let image_device_path: &DevicePath = &image_device_path_protocol;
+
+    // Get the file-path portion of the device path which is typically behind
+    // device path node (0x4, 0x4). The string is in upper case.
+
+    // TODO we need an abstraction in the UEFI crate for this.
+    let image_device_path_file_path = image_device_path
+        .node_iter()
+        .find(|node| {
+            node.device_type() == DeviceType::MEDIA /* 0x4 */
+                && node.sub_type() == DeviceSubType::HARDWARE_VENDOR /* 0x4 */
+        })
+        .map(|node| {
+            CStr16::from_u16_with_nul(node.data_u16().unwrap()).expect("should be valid CCtr16")
+        })
+        // to Rust string
+        .map(|cstr16| cstr16.to_string().to_uppercase())
+        .expect("should have file-path portion in device path");
```

Comment:
For extracting the file path, try something like this (haven't actually tested it, but I think this should work):

```rust
    let image_device_path_file_path = image_device_path
        .node_iter()
        .find_map(|node| {
            let node: &FilePath = node.try_into().ok()?;
            let path = node.path_name().to_cstring16().ok()?;
            Some(path.to_string().to_uppercase())
        })
        .expect("should have file-path portion in device path");
```

Reply:
thanks, that's a much better approach!